frontend/public/components/dashboard/details-card/detail-item.tsx

```diff
@@ -1,19 +1,27 @@
 import * as React from 'react';
-import classNames from 'classnames';
-
 import { LoadingInline } from '../../utils';
 
-export const DetailItem: React.FC<DetailItemProps> = React.memo(({ title, value, isLoading }) => (
-  <React.Fragment>
-    <dt className="co-details-card__item-title">{title}</dt>
-    <dd className={classNames('co-details-card__item-value', {'text-secondary': !value})}>
-      {isLoading ? <LoadingInline /> : value || 'Unavailable'}
-    </dd>
-  </React.Fragment>
-));
+export const DetailItem: React.FC<DetailItemProps> = React.memo(
+  ({ title, isLoading, children, error = false }) => {
+    let status: React.ReactNode;
+    if (isLoading) {
+      status = <LoadingInline />;
+    } else if (error) {
+      status = <span className="text-secondary">Unavailable</span>;
+    } else {
+      status = children;
+    }
+    return (
+      <React.Fragment>
+        <dt className="co-details-card__item-title">{title}</dt>
+        <dd className="co-details-card__item-value">{status}</dd>
+      </React.Fragment>
+    );
+  }
+);
 
-type DetailItemProps = {
+type DetailItemProps = React.PropsWithChildren<{
```

Comment:
```suggestion
type DetailItemProps = Required<React.PropsWithChildren<{
```
@jtomasek thank you for pointing that out. In this case, children is a required item. So maybe I should change it to this? @christianvogt what do you think?

Reply:
Console defines required children in many places like this:
```suggestion
type DetailItemProps = {
  children: React.ReactNode;
```